src/java.scripting/share/classes/javax/script/SimpleScriptContext.java

```diff
@@ -95,7 +97,7 @@ public SimpleScriptContext() {
     }
 
     private static InputStreamReader stdinReader() {
-        Charset charset = Charset.forName(System.getProperty("stdin.encoding"), Charset.defaultCharset());
+        Charset charset = Charset.forName(StaticProperty.stdinEncoding(), Charset.defaultCharset());
```

Comment:
This is the jrunscript command line tool. I suppose it's possible it could run a script that changes stdin.encoding.